app/models/social_link.rb

```diff
@@ -0,0 +1,67 @@
+# == Schema Information
+#
+# Table name: social_links
+#
+#  id         :bigint(8)        not null, primary key
+#  user_id    :bigint(8)        not null
+#  url        :string           not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+# Indexes
+#
+#  index_social_links_on_user_id  (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => users.id)
+#
+
+class SocialLink < ApplicationRecord
+  belongs_to :user
+
+  validates :url, :presence => true, :format => { :with => URI::DEFAULT_PARSER.make_regexp(%w[http https]), :message => I18n.t("profiles.edit.social_links.http_parse_error") }
```

Comment:
This doesn't check if `url` starts with `http://`.

And if you also disable CSP:
![image](https://github.com/user-attachments/assets/52f7c65d-f423-4093-8483-63e236e0e6df)